### PR TITLE
remove isArray, filter, reject, isNull

### PR DESCRIPTION
**Pull request by bkendall**
### Description

removes `isArray`, `filter`, `reject`, and `isNull`

Also fixes a bug where the `checkValidTargetFilters` was checking _every character in the only string_ instead of each thing split by a comma. That was interesting to find.

### CODE REVIEW

File: src/checkValidTargetFilters.ts [hunk 1]
```diff
@@ -24,10 +22,10 @@ export function checkValidTargetFilters(options: Options): Promise<void> {
     if (options.except) {
       return reject(new FirebaseError("Cannot specify both --only and --except"));
     }
-    if (targetContainsFilter(["database", "storage", "hosting"])) {
+    if (targetContainsFilter(["database", "storage"])) {
```
Comment (joehan) [inline src/checkValidTargetFilters.ts:25]: This logic seems outdated - shouldn't this trigger for all VALID_TARGETS except functions & hosting? 

alid targets are:   
"database",
  "storage",
  "firestore",
  "functions",
  "hosting",
  "remoteconfig",
  "extensions",

**Review by joehan — CHANGES_REQUESTED** [review]
Found a logic error that we should fix while were here

File: src/checkValidTargetFilters.ts (hunk 1 above)
Comment (bkendall) [reply src/checkValidTargetFilters.ts]: I've revamped this file to be a little better to understand. Let me know what you think.

File: src/checkValidTargetFilters.ts [hunk 2]
```diff
@@ -1,37 +1,65 @@
...
       return reject(
         new FirebaseError(
-          "Filters specified with colons (e.g. --only functions:func1,functions:func2) are only supported for functions"
+          "Filters specified with colons (e.g. --only functions:func1,functions:func2) are only supported for functions and hosting"
         )
       );
     }
-    if (targetContainsFilter(["functions"]) && targetDoesNotContainFilter(["functions"])) {
+    const targetsForFunctions = targetsForTypes(only, "functions");
+    console.error(
```
Comment (joehan) [inline src/checkValidTargetFilters.ts:52]: Remove this

**Review by joehan — APPROVED** [review]
LGTM except for the leftover console.error

**Pull request merged**